### PR TITLE
docs: workflow and linter improvements

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

- [x] The `link-check` workflow is now only activated upon `.md` file modifications. (excluding `.github/` directory)~~
- [x] New `markdown-lint` workflow has been added
  - We've shifted from  [`markdownlint`](https://github.com/DavidAnson/markdownlint) to [`markdownlint-cli2`](https://github.com/DavidAnson/markdownlint-cli2) in this repo. (Makefile is updated accordingly)
  - This workflow only runs on modified markdown files if a `.md` file has been modified (not including `.github/` directory).
  - There is a new config file for this tool at [`.markdownlint-cli2.jsonc`](https://github.com/cosmos/ibc-go/pull/3577/files#diff-4929acf5ce7500ed4d3f00992a1c918e702427b7857fed53152e48fd58c2ae9c), this tool also continues to use [`.markdownlint.jsonc`](https://github.com/cosmos/ibc-go/blob/serdar/issue%233522-store-docs-for-past-versions/.markdownlint.jsonc)
  - Additional rules are added to [`.markdownlint.jsonc`](https://github.com/cosmos/ibc-go/blob/serdar/issue%233522-store-docs-for-past-versions/.markdownlint.jsonc) to address: #3543.

closes: #3543 

### Draft PR to Open PR Requirements

- [x] #3577 is merged

### Commit Message / Changelog Entry

```text
docs: workflow and linter improvements
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by charleenfei — COMMENTED** [review]
there are some redundant changes in this PR and the other docs PR right? if so, maybe worth merging the other first before this one to make the review easier 🙈

File: Makefile [hunk 1]
```diff
@@ -138,16 +138,6 @@ go.sum: go.mod
 ###                              Documentation                              ###
 ###############################################################################
 
-update-swagger-docs: statik
```
Comment (crodriguezvega) [inline Makefile:141]: This is not used?

File: Makefile [hunk 2]
```diff
@@ -292,10 +282,13 @@ format:
...
 docs-lint:
-	markdownlint . --fix
+	markdownlint-cli2 "**.md"
+
+docs-lint-fix:
+	markdownlint-cli2-fix "**.md"
 
-docs-lint-changed:
-	./scripts/linting/lint-changed-md-files.sh
+docs-link-check:
```
Comment (crodriguezvega) [inline Makefile:290]: To be able to run this we need to have `markdown-link-check` installed, right?

**Review by crodriguezvega — APPROVED** [review]
Looks good, @srdtrk! I left a couple of questions. And I pushed couple of `md` files that were listed when I run the makefile command.

File: docs/docs/01-ibc/09-roadmap.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ slug: /roadmap/roadmap
 
 # Roadmap ibc-go
 
-_Lastest update: June 23th, 2023_
+*Lastest update: June 23th, 2023*
```
Comment (charleenfei) [inline docs/docs/01-ibc/09-roadmap.md:10]:
```suggestion
*Latest update: June 23th, 2023*
```

**Review by charleenfei — APPROVED** [review]
just one nit, otherwise LGTM!

**Pull request merged**